ui-admin/src/portal/MailingListView.tsx

```diff
@@ -83,17 +83,10 @@ export default function MailingListView({ portalContext, portalEnv }:
   }
   const numSelected = Object.keys(rowSelection).length
 
-  const loadMailingList = () => {
-    setIsLoading(true)
-    setRowSelection({})
-    Api.fetchMailingList(portalContext.portal.shortcode, portalEnv.environmentName).then(result => {
-      setContacts(result)
-      setIsLoading(false)
-    }).catch((e: Error) => {
-      alert(`error loading mailing list ${  e}`)
-      setIsLoading(false)
-    })
-  }
+  const { isLoading, reload } = useLoadingEffect(async () => {
+    const result = await Api.fetchMailingList(portalContext.portal.shortcode, portalEnv.environmentName)
+    setContacts(result)
+  }, [portalContext.portal.shortcode, portalEnv.environmentName])
```

Comment:
I think this looks really nice! It's a common pattern, so definitely a good call to extract it.